src/models/CommonModel.ts

```diff
@@ -471,6 +471,16 @@ export class CommonModel {
           mergeFrom,
           originalInput
         );
+
+        if (
+          mergeTo.properties[String(propName)].$id?.includes(
+            'anonymous_schema'
+          ) &&
+          !propValue.$id?.includes('anonymous_schema')
+        ) {
+          mergeTo.properties[String(propName)].$id = propValue.$id;
+        }
```

Comment:
Instead of here, I think it should be added to the mergeCommonModels function as it's really there you want to selectively decide whether mergeFrom $id is better than the mergeTo is: https://github.com/asyncapi/modelina/blob/7a8197370a0c679f588e98f471558cb766b19dc4/src/models/CommonModel.ts#L773

```ts
    // Base the $id of the model on the most suited one.
    // If mergeTo is using anonymous_schema id, and the mergeFrom does not, use the mergeFrom $id.
    if (
      mergeTo.$id?.includes(
        'anonymous_schema'
      ) &&
      !mergeFrom.$id?.includes('anonymous_schema')
    ) {
      mergeTo.$id = mergeFrom.$id;
    } else {
      mergeTo.$id = mergeTo.$id || mergeFrom.$id;
    }
```

Reply:
Makes sense. Done